dev/tests/api-functional/testsuite/Magento/GraphQl/Customer/DeleteCustomerTest.php

```diff
@@ -0,0 +1,106 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\GraphQl\Customer;
+
+use Magento\Customer\Api\CustomerRepositoryInterface;
+use Magento\Framework\Exception\AuthenticationException;
+use Magento\TestFramework\Helper\Bootstrap;
+use Magento\TestFramework\TestCase\GraphQlAbstract;
+use Magento\Integration\Api\CustomerTokenServiceInterface;
+
+/**
+ * Delete custom tests
+ */
+class DeleteCustomerTest extends GraphQlAbstract
+{
+    /**
+     * @var CustomerTokenServiceInterface
+     */
+    private $customerTokenService;
+
+    /**
+     * @var CustomerRepositoryInterface
+     */
+    private $customerRepository;
+
+    /**
+     * @var LockCustomer
+     */
+    private $lockCustomer;
+
+    /**
+     * @inheritdoc
+     */
+    protected function setUp(): void
+    {
+        parent::setUp();
+        $this->customerTokenService = Bootstrap::getObjectManager()->get(CustomerTokenServiceInterface::class);
+        $this->customerRepository = Bootstrap::getObjectManager()->get(CustomerRepositoryInterface::class);
+        $this->lockCustomer = Bootstrap::getObjectManager()->get(LockCustomer::class);
+    }
+
+    /**
+     * @magentoApiDataFixture Magento/Customer/_files/customer.php
+     */
+    public function testDeleteCustomer()
+    {
+        $response = $this->graphQlMutation($this->getMutation(), [], '', $this->getHeaderMap());
+        $this->assertArrayHasKey('deleteCustomer', $response);
+        $this->assertTrue($response['deleteCustomer']);
+    }
+
+    /**
+     */
+    public function testDeleteCustomerIfUserIsNotAuthorized()
+    {
+        $this->expectException(\Exception::class);
+        $this->expectExceptionMessage('The current customer isn\'t authorized.');
+        $this->graphQlMutation($this->getMutation());
+    }
+
+    /**
+     * @magentoApiDataFixture Magento/Customer/_files/customer.php
+     * @magentoApiDataFixture Magento/Customer/_files/customer_two_addresses.php
+     */
+    public function testDeleteCustomerIfAccountIsLocked()
+    {
+        $this->lockCustomer->execute(1);
+        $response = $this->graphQlMutation($this->getMutation(), [], '', $this->getHeaderMap());
+        $this->assertArrayHasKey('deleteCustomer', $response);
+        $this->assertTrue($response['deleteCustomer']);
```

Comment:
You are right. We can't log in by the locked customer 
```
message": "The account sign-in was incorrect or your account is disabled temporarily. Please wait and try again later.",
```
This test was successful because the customer is used from the registry during authentication
`$this->customerRegistry->retrieve($customerId)` and there customer is saved as unlocked. 

This test failed if I use the `Magento/Customer/_files/locked_customer.php` fixture instead of 
```
Magento/Customer/_files/customer.php
 $this->lockCustomer->execute(1);
```
I will fix it. 
Thanks!